pkg/chunk/chunk_store.go

```diff
@@ -286,12 +289,12 @@ func (c *baseStore) validateQueryTimeRange(ctx context.Context, userID string, f
 	defer log.Span.Finish()
 
 	if *through < *from {
-		return false, httpgrpc.Errorf(http.StatusBadRequest, "invalid query, through < from (%s < %s)", through, from)
+		return false, QueryError(fmt.Sprintf("invalid query, through < from (%s < %s)", through, from))
```

Comment:
Not using `httpgrpc.Errorf()` in `validateQueryTimeRange()` changes the behaviour for `LabelValuesForMetricName()` and `LabelNamesForMetricName()` where we want to return an `httpgrpc` error. An option could be always wrapping the error there.

Reply:
`LabelValuesForMetricName` and `LabelNamesForMetricName` are not even used in Cortex :) (but they are in Loki).

I think Loki should recognize QueryError and convert it to appropriate error in the layer it is using. Store itself should not return gRPC errors.

Reply:
I will make sure that it is handled correctly in Loki, if this PR is merged.